table/block_based/block_based_table_reader.cc

```diff
@@ -1967,7 +1967,8 @@ CachableEntry<FilterBlockReader> BlockBasedTable::GetFilter(
     }
   }
 
-  if (block_cache_tracer_ && lookup_context) {
+  if (block_cache_tracer_ && lookup_context &&
+      block_cache_tracer_->is_tracing_enabled()) {
```

Comment:
Depending on which one is more likely to be false, to me it's either `block_cache_tracer_ && lock_cache_tracer_->is_tracing_enabled() && lookup_context ` or `lookup_context && block_cache_tracer_ && lock_cache_tracer_->is_tracing_enabled() `

Reply:
I updated the order of the checks to `block_cache_tracer_ && block_cache_tracer_->is_tracing_enabled() && lookup_context`